common/src/java/org/apache/hadoop/hive/conf/HiveConf.java

```diff
@@ -770,6 +770,8 @@ private static void populateLlapDaemonVarsSet(Set<String> llapDaemonVarsSetLocal
         "If not set, defaults to the codec extension for text files (e.g. \".gz\"), or no extension otherwise."),
 
     HIVE_IN_TEST("hive.in.test", false, "internal usage only, true in test mode", true),
+    HIVE_IN_TEST_ICEBERG("hive.in.iceberg.test", false, "internal usage only, true when " +
```

Comment:
why is this needed?

Reply:
During the creation of the iceberg table an exclusive lock is requested on the hms table. This logic is guarded by the value of `HIVE_IN_TEST` and `HINVE_IN_TEZ_TEST`. If any of these values are true and the operationtype is unset, we get an exception, which is the case in q tests. 
```java
f (lc.isSetOperationType() && lc.getOperationType() == DataOperationType.UNSET &&
                (MetastoreConf.getBoolVar(conf, ConfVars.HIVE_IN_TEST) || MetastoreConf.getBoolVar(conf, ConfVars.HIVE_IN_TEZ_TEST))) {
          throw new IllegalStateException("Bug: operationType=" + lc.getOperationType() + " for component "
                  + lc + " agentInfo=" + rqst.getAgentInfo());
        }
```
Changing the `HIVE_IN_TEST` to false is not the right solution, because it is evaluated in many other places as well, activating unwanted code paths.

Reply:
How I hate test code in the production codepath, I can not tell you...
@deniskuzZ: How hard would it be to get rid of this?

Reply:
This caused trouble when migrating the iceberg-mr code to the new iceberg-handler module. The TestHiveShell configs had to patched to get around this as well..